src/jdk.incubator.vector/linux/native/libvmath/vect_math.c

```diff
@@ -0,0 +1,121 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include <sleef.h>
+
+#define DEFINE_VECTOR_MATH_UNARY(op, type, cpu) \
+type op##cpu(type input) {                      \
+  return Sleef_##op##cpu(input);                \
+}
+
+DEFINE_VECTOR_MATH_UNARY(tanf4_u10,   float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(tanhf4_u10,  float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(sinf4_u10,   float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(sinhf4_u10,  float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(cosf4_u10,   float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(coshf4_u10,  float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(asinf4_u10,  float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(acosf4_u10,  float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(atanf4_u10,  float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(cbrtf4_u10,  float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(logf4_u10,   float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(log10f4_u10, float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(log1pf4_u10, float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(expf4_u10,   float32x4_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(expm1f4_u10, float32x4_t, advsimd)
+
+DEFINE_VECTOR_MATH_UNARY(tand2_u10,   float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(tanhd2_u10,  float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(sind2_u10,   float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(sinhd2_u10,  float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(cosd2_u10,   float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(coshd2_u10,  float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(asind2_u10,  float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(acosd2_u10,  float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(atand2_u10,  float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(cbrtd2_u10,  float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(logd2_u10,   float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(log10d2_u10, float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(log1pd2_u10, float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(expd2_u10,   float64x2_t, advsimd)
+DEFINE_VECTOR_MATH_UNARY(expm1d2_u10, float64x2_t, advsimd)
+
+#ifdef __ARM_FEATURE_SVE
```

Comment:
No, we're building a standard binary that will use SVE if it's present on the machine it's running on. Such compile-time feature tests are inappropriate. You need to provide SVE entry points for the target machine, no matter how HotSpot is configured and built.